modules/programs/zsh.nix

```diff
@@ -198,9 +210,10 @@ in
       home.file.".zshenv".text = ''
         ${optionalString (config.home.sessionVariableSetter == "zsh")
           envVarsStr}
+        ${optionalString (cfg.dotDir != "") (export "ZDOTDIR" ("$HOME/" + cfg.dotDir))}
```

Comment:
Maybe this should also be added to `home.sessionVariables`, so that e.g. calling zsh from bash as the default shell works correctly.